geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/iterators/StatsIterator.scala

```diff
@@ -0,0 +1,145 @@
+import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes.buildTypeName
+import org.locationtech.geomesa.utils.geotools.{GeometryUtils, SimpleFeatureTypes}
+import org.locationtech.geomesa.utils.stats._
+import org.opengis.feature.simple.SimpleFeatureType
+
+/**
+ * Reads simple features and observe them with a Stat server-side
+ *
+ * @param other
+ * @param env
+ */
+class StatsIterator(other: StatsIterator, env: IteratorEnvironment)
+  extends FeatureAggregatingIterator[StatsIteratorResult](other, env) {
+
+  import org.locationtech.geomesa.accumulo.iterators.StatsIterator.STATS_ITERATOR_SFT_STRING
+
+  var stat: Stat = null
+
+  projectedSFTDef = STATS_ITERATOR_SFT_STRING
+
+  def this() = this(null, null)
+
+  override def initProjectedSFTDefClassSpecificVariables(source: SortedKeyValueIterator[Key, Value],
+                                                         options: JMap[String, String],
+                                                         env: IteratorEnvironment): Unit = {
+    val statString = StatsIterator.getStatsString(options)
+    stat = Stat(simpleFeatureType, statString)
+  }
+
+  override def handleKeyValue(resultO: Option[StatsIteratorResult],
+                              topSourceKey: Key,
+                              topSourceValue: Value): StatsIteratorResult = {
+    val sf = originalDecoder.deserialize(topSourceValue.get())
+    val result = resultO.getOrElse(StatsIteratorResult(stat))
+    result.stat.observe(sf)
+    result
+  }
+}
+
+object StatsIterator extends Logging {
+  val STATS_STRING_KEY = "geomesa.stats.string"
+  val STATS_FEATURE_TYPE_KEY = "geomesa.stats.featuretype"
+  val STATS = "stats"
+  val STATS_ITERATOR_SFT_STRING = s"$STATS:String,geom:Geometry"
+
+  /**
+   * Adds key-value pairs consisting of the statString and featureType to the cfg client-side
+   * The config is then read server-side to set up the iterator
+   *
+   * @param cfg iterator config settings
+   * @param statString stat query hint string (e.g. "MinMax(foo)")
+   */
+  def configure(cfg: IteratorSetting, statString: String) = {
+    setStatsString(cfg, statString)
+  }
+
+  def setStatsString(iterSettings: IteratorSetting, statString: String): Unit = {
+    iterSettings.addOption(STATS_STRING_KEY, statString)
+  }
+
+  def getStatsString(options: JMap[String, String]): String = {
+    options.get(STATS_STRING_KEY)
+  }
+
+  def createFeatureType(origFeatureType: SimpleFeatureType) = {
+    //Need a filler namespace, else geoserver throws nullptr exception for xml output
+    val (namespace, name) = buildTypeName(origFeatureType.getTypeName)
+    val outNamespace = if (namespace == null){
+        "NullNamespace"
+      } else {
+        namespace
+      }
+    SimpleFeatureTypes.createType(outNamespace, name, StatsIterator.STATS_ITERATOR_SFT_STRING)
+  }
+
+  def combineStats(stat1: Stat, stat2: Stat): Stat = {
+    stat1.add(stat2)
+  }
+
+  def encodeStat(stat: Stat): String = {
```

Comment:
This is a holdover from the TemporalDensityIterator that I maintained, but do not really understand the purpose of. There were two ways to serialize the TemporalDensityIterator as a String:

1) Encode to byte array and stringify
2) Encode to JSON

I suppose #1 is faster and cheaper in terms of memory, but was that the only reason two methods of serializing to a string were used?